src/rest/StreamEndpoints.js

```diff
@@ -106,6 +108,12 @@ export function produceToStream(streamOrId, data, apiKey = this.options.apiKey,
         `${this.options.restUrl}/streams/${streamId}/data`,
         apiKey,
         {
+            httpAgent: http.Agent({
```

Comment:
A few questions regarding this fix:

- Is this creating a new `Agent` for every request? (Here, actually **two** agents for every request). I would assume that the same `Agent` is meant to be shared across many requests.

- I'm not sure what exactly this line does. [The docs](https://nodejs.org/api/http.html#http_class_http_agent) show agent creation which includes the `new` keyword.